cmake/common/osconfig.cmake

```diff
@@ -0,0 +1,17 @@
+# CMake operating system bootstrap module
+
+include_guard(GLOBAL)
+
+# Set minimum CMake version specific to host operating system, add OS-specific module directory to default search paths,
+# and set helper variables for OS detection in other CMake list files.
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Windows")
+  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/windows")
+  set(OS_WINDOWS TRUE)
+elseif(CMAKE_HOST_SYSTEM_NAME STREQUAL "Darwin")
+  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/macos")
+  set(OS_MACOS TRUE)
+elseif(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux|FreeBSD|OpenBSD")
+  list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/linux")
+  string(TOUPPER "${CMAKE_HOST_SYSTEM_NAME}" _SYSTEM_NAME_U)
+  set(OS_${_SYSTEM_NAME_U} TRUE)
+endif()
```

Comment:
Don't enable/add cross-compile if you rely on `CMAKE_HOST_` variable.

Reply:
That is actually the desired behaviour here - the decisions need to be made based on the _host_ architecture, not the _target_ architecture. We do not support cross compiling for other platforms, only for other architectures on the same platform.

Cross compilation with CMake should _never_ change the host variables (hence why they are called `CMAKE_HOST_<SOMETHING>`.

Reply:
I'm just saying that `CMAKE_SYSTEM_NAME` must be used there. It would makes no sense to use the host system name while cross-compiling, the toolchain set `CMAKE_SYSTEM_NAME` for a reason.

Reply:
It can set it all it wants, the bootstrapper _requires_ the host system. It is a conscious choice and not a mistake. Which is also why the bootstrapper runs _before_ the project call - at this point in execution there is no notion of a project or a compilation target, just the host system the CMake script runs on.

Reply:
So you voluntarily prevent people to be able to test-compile for other platform (not create a bundle just test if it compiles) through cross-compilation.